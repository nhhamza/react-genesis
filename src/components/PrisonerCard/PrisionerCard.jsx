import React from 'react';
import PropTypes from 'prop-types';
import ArrowForwardIosIcon from '@material-ui/icons/ArrowForwardIos';

const CardPrisioner = ({ card }) => {
  return (
    <div
      key={card.id}
      title="result__container__card"
      className="result__container__card"
    >
      <div>
        <img
          className="result__container__card__image"
          src={card?.creator?.avatarUrl}
          alt={card.title}
        />
      </div>
      <div className="result__container__card__body">
        <div className="result__container__card__body__phone">
          {card?.creator.phone}
        </div>
        <div className="result__container__card__body__title">
          {card?.creator.name}
        </div>
        <div className="result__container__card__body__date">{card?.date}</div>
      </div>
      <div className="result__container__card__navigation">
        <ArrowForwardIosIcon fontSize="small" />
      </div>
    </div>
  );
};

CardPrisioner.propTypes = {
  card: PropTypes.objectOf({})
};

export default CardPrisioner;
