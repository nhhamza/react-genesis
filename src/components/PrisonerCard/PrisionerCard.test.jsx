import React from 'react';
import { shallow } from 'enzyme';
import PrisionerCard from './PrisionerCard.jsx';

describe('PrisionerCard component', () => {
  const MOCK_PROPS = {
    card: {
      id: 'id',
      title: 'title',
      creator: {}
    }
  };

  const wrapper = shallow(<PrisionerCard {...MOCK_PROPS} />);
  it('should render', () => {
    expect(wrapper.exists()).toMatchSnapshot();
  });
});
