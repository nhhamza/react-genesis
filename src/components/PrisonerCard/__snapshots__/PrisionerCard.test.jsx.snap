// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrisionerCard component should render 1`] = `true`;
